lms/djangoapps/courseware/module_render.py

```diff
@@ -178,13 +182,55 @@ def toc_for_course(user, request, course, active_chapter, active_section, field_
                           section.url_name == active_section)
 
                 if not section.hide_from_toc:
-                    sections.append({'display_name': section.display_name_with_default,
-                                     'url_name': section.url_name,
-                                     'format': section.format if section.format is not None else '',
-                                     'due': section.due,
-                                     'active': active,
-                                     'graded': section.graded,
-                                     })
+                    section_context = {
+                        'display_name': section.display_name_with_default,
+                        'url_name': section.url_name,
+                        'format': section.format if section.format is not None else '',
+                        'due': section.due,
+                        'active': active,
+                        'graded': section.graded,
+                    }
+
+                    #
+                    # Add in rendering context for proctored exams
+                    # if applicable
+                    #
+                    is_proctored_enabled = (
+                        getattr(section, 'is_proctored_enabled', False) and
+                        settings.FEATURES.get('ENABLE_PROCTORED_EXAMS', False)
+                    )
+                    if is_proctored_enabled:
+                        #
+                        # call into edx_proctoring subsystem
+                        # to get relevant proctoring information regarding this
+                        # level of the courseware
+                        #
+                        # This will return None, if (user, course_id, content_id)
+                        # is not applicable
+                        #
+                        proctoring_attempt_context = None
+                        try:
+                            proctoring_attempt_context = get_attempt_status_summary(
+                                user.id,
+                                unicode(course.id),
+                                unicode(section.location)
+                            )
+                        except Exception, ex:  # pylint: disable=broad-except
```

Comment:
yea, I hear ya. I added this to gracefully fail this small element of the feature (to render status indicators). This area of the code is very core to the application, so my preference is to keep this here and - once we get some production deploy cycles on it - remove the catch-and-continue pattern.